flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/TableSourceTest.scala

```diff
@@ -130,6 +131,60 @@ class TableSourceTest extends TableTestBase {
     util.verifyPlan(sqlQuery)
   }
 
+
+  @Test
+  def testLegacyRowTimeTableGroupWindow(): Unit = {
+    util.tableEnv.connect(new ConnectorDescriptor("TestTableSourceWithTime", 1, false) {
+      override protected def toConnectorProperties: JMap[String, String] = {
+        Collections.emptyMap()
+      }
```

Comment:
Can we have a dedicated descriptor for `TestTableSourceWithTime`? This code looks confusing.

Reply:
I don't think so. A dedicated descriptor for every custom table source is wasteful and should be avoid. Maybe we can use `CustomConnectorDescriptor` after we port it in FLINK-16029?

Reply:
Introducing a generic connector descriptor is another topic, and I'm concerned `CustomConnectorDescriptor` is not easy-to-use enough. 

Maybe we can introduce a general `TestConnectorDescriptor` in tests.